src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -524,28 +523,78 @@ internal void WriteCentralDirectoryFileHeader()
                 extraFieldLength = (ushort)bigExtraFieldLength;
             }
 
-            writer.Write(ZipCentralDirectoryFileHeader.SignatureConstant);      // Central directory file header signature  (4 bytes)
-            writer.Write((byte)_versionMadeBySpecification);                    // Version made by Specification (version)  (1 byte)
-            writer.Write((byte)CurrentZipPlatform);                             // Version made by Compatibility (type)     (1 byte)
-            writer.Write((ushort)_versionToExtract);                            // Minimum version needed to extract        (2 bytes)
-            writer.Write((ushort)_generalPurposeBitFlag);                       // General Purpose bit flag                 (2 bytes)
-            writer.Write((ushort)CompressionMethod);                            // The Compression method                   (2 bytes)
-            writer.Write(ZipHelper.DateTimeToDosTime(_lastModified.DateTime));  // File last modification time and date     (4 bytes)
-            writer.Write(_crc32);                                               // CRC-32                                   (4 bytes)
-            writer.Write(compressedSizeTruncated);                              // Compressed Size                          (4 bytes)
-            writer.Write(uncompressedSizeTruncated);                            // Uncompressed Size                        (4 bytes)
-            writer.Write((ushort)_storedEntryNameBytes.Length);                 // File Name Length                         (2 bytes)
-            writer.Write(extraFieldLength);                                     // Extra Field Length                       (2 bytes)
+            Span<byte> cdStaticHeader = stackalloc byte[ZipCentralDirectoryFileHeader.BlockConstantSectionSize];
+            int cdPosition = 0;
 
+            // Central directory file header signature  (4 bytes)
+            ZipCentralDirectoryFileHeader.SignatureConstantBytes.CopyTo(cdStaticHeader);
+            cdPosition += ZipCentralDirectoryFileHeader.SignatureConstantBytes.Length;
+
+            // Version made by Specification (version)  (1 byte)
+            cdStaticHeader[cdPosition++] = (byte)_versionMadeBySpecification;
+
+            // Version made by Compatibility (type)     (1 byte)
+            cdStaticHeader[cdPosition++] = (byte)CurrentZipPlatform;                        // Version made by Compatibility (type)     (1 byte)
```

Comment:
The comment doesn't need to be there twice, at minimum.

```suggestion
            // Version made by Compatibility (type)     (1 byte)
            cdStaticHeader[cdPosition++] = (byte)CurrentZipPlatform;
```

Reply:
Thanks - I missed that comment for some reason, but have checked the rest; it's the only one.